app/models/ability.rb

```diff
@@ -3,30 +3,52 @@ class Ability
 
   # Initializes the ability class
   def initialize(user)
-    # Order Abilities
-    # (Check https://github.com/CanCanCommunity/cancancan/wiki/Ability-Precedence)
-    # Check roles of user, using rolify. Role name is *case sensitive*
-    # user.is_organizer? or user.has_role? :organizer
-    # user.is_cfp_of? Conference or user.has_role? :cfp, Conference
-    # user.is_info_desk_of? Conference
-    # user.is_volunteers_coordinator_of? Conference
-    # user.is_attendee_of? Conference
-    # The following is wrong because a user will only have 'cfp' role for a specific conference
-    # user.is_cfp? # This is always false
-
     user ||= User.new
 
-    # This is what sets up the different abilities
     if user.new_record?
       not_signed_in
-    # Checks if the user does not have any role and is not an admin
     elsif user.roles.any? || user.is_admin
-      signed_in_with_roles(user)
+      common_abilities_for_admins(user)
     else
       signed_in(user)
     end
   end
 
+  # Abilities for users with roles wandering around in non-admin views.
+  def common_abilities_for_admins(user)
+    signed_in(user)
+    conf_ids_for_organizer = Conference.with_role(:organizer, user).pluck(:id)
+    conf_ids_for_cfp = Conference.with_role(:cfp, user).pluck(:id)
+    conf_ids_for_info_desk = Conference.with_role(:info_desk, user).pluck(:id)
+
+    if conf_ids_for_organizer
+
+      # To access splashpage of their conference if it is not public
+      can :show, Conference, id: conf_ids_for_organizer
+
+      # To access conference/proposals/registrations
+      can :manage, Registration, conference_id: conf_ids_for_organizer
+
+      # To access conference/proposals
+      can :manage, Event, program: { conference_id: conf_ids_for_organizer }
+
+      # To access comment link in menu bar
+      can :index, Comment, commentable_type: 'Event',
+                           commentable_id: Event.where(program_id: Program.where(conference_id: conf_ids_for_organizer).pluck(:id)).pluck(:id)
+    elsif conf_ids_for_cfp
+
+      can :index, Comment, commentable_type: 'Event',
+                           commentable_id: Event.where(program_id: Program.where(conference_id: conf_ids_for_cfp).pluck(:id)).pluck(:id)
+      can :manage, Event, program: { conference_id: conf_ids_for_cfp }
+
+    elsif conf_ids_for_info_desk
+      can :manage, Registration, conference_id: conf_ids_for_info_desk
+    end
```

Comment:
I don't think it's right that these are `if-elsif` conditions.
For example, a user can have `conf_ids_for_organizer == [1]` and `conf_ids_for_info_desk == [2]` (so they're an organizer of one conference, but an info-desk for another conference).
So with this code as it is, they will get all organizer abilities for Conf1 but they will NOT get the info desk permission for Conf2